test/e2e/metrics_test.go

```diff
@@ -15,16 +15,33 @@ package e2e
 
 import (
 	"bytes"
+	"fmt"
 	"io"
 	"os/exec"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/require"
 
 	"github.com/operator-framework/operator-controller/test/utils"
 )
 
+// retry attempts to run fn up to attempts times with delay between attempts.
+func retry(fn func() error) error {
```

Comment:
There are helper functions in a kubernetes `wait` package that we should use (if we actually need this). I think it is in apimachinery.

Reply:
I backed these changes out in most recent commit